model/device.go

```diff
@@ -122,6 +122,41 @@ type Device struct {
 	UpdatedTs time.Time `json:"updated_ts" bson:"updated_ts,omitempty"`
 }
 
+// internalDevice is only used internally to avoid recursive type-loops for
+// member functions.
+type internalDevice Device
+
+func (d *Device) UnmarshalBSON(b []byte) error {
+	if err := bson.Unmarshal(b, (*internalDevice)(d)); err != nil {
+		return err
+	}
+	for _, attr := range d.Attributes {
+		if attr.Scope == "identity" {
```

Comment:
see the other comment, we do have const for that. it probably happens more times.